std/concurrency.d

```diff
@@ -2403,3 +2403,118 @@ version( unittest )
         scheduler = null;
     }
 }
+
+//////////////////////////////////////////////////////////////////////////////
+// initOnce
+//////////////////////////////////////////////////////////////////////////////
+
+private template initOnceLock()
+{
+    __gshared Mutex lock;
+
+    shared static this()
+    {
+        lock = new Mutex;
+    }
+
+    @property Mutex initOnceLock()
+    {
+        return lock;
+    }
+}
+
+/**
+ * Initializes $(D_PARAM var) with the lazy $(D_PARAM init) value in a
+ * thread-safe manner.  The implementation guarantees that all threads
+ * simultaneously calling initOnce with the same $(D_PARAM var) argument block
+ * until $(D_PARAM var) is fully initialized. All side-effects of $(D_PARAM init)
+ * are globally visible afterwards.
+ *
```

Comment:
Done, but I still disagree.
We don't improve our documentation by adding noise.
What's that supposed to tell me?
http://www.cplusplus.com/reference/cmath/pow/#parameters